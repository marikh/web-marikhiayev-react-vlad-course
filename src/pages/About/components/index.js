import AboutHeroArea from './AboutHeroArea';

export {
    AboutHeroArea
}
