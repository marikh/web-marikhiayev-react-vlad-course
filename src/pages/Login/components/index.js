import LoginHeroArea from './LoginHeroArea';

export {
    LoginHeroArea
}
