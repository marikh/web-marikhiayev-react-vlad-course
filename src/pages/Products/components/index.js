import ProductsHeroArea from './ProductsHeroArea';

export {
    ProductsHeroArea
}
