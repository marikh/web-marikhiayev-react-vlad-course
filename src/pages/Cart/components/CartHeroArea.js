import React from 'react';
import { HeroArea, Heading } from '../../../components';

export default () => (
    <HeroArea>
        <Heading size={3}>
            Cart
        </Heading>
    </HeroArea>
) 