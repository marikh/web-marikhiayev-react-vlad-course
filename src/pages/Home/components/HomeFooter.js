import React from 'react';
import { SocialLinks } from '../../../components/';
import { withRouter } from 'react-router-dom';

export default withRouter(({ links, history, location, match }) => {
    return (
        <div>
            <SocialLinks links={links} />
            <h3 onClick={() => history.goBack()}>Back</h3>
        </div>
    )
})

