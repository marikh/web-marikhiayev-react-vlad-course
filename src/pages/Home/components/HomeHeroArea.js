import React from 'react';
import { HeroArea, Heading } from '../../../components/';

export default () => (
    <HeroArea>
        <Heading size={1}>
            Home Page
        </Heading>
    </HeroArea>
)
