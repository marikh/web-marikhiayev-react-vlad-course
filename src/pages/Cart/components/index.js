import CartHeroArea from './CartHeroArea';

export {
    CartHeroArea
}
