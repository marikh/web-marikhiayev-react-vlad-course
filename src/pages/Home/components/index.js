import HomeFooter from './HomeFooter';
import HomeHeroArea from './HomeHeroArea';

export {
    HomeFooter,
    HomeHeroArea    
}
