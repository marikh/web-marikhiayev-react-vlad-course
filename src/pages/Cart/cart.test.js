import { mount, shallow } from "enzyme";
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import { Layout } from '../../components/';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import ConnectedCart from '../../pages/Cart/';
import {createStore} from 'redux';

const productsIDs = [
    '123123-234-2341-123123-123123',
    '123123-12342-456456-123123-123123',
];
const MENU_INITIAL_STATE = {showProtectedLinks : false};
const CART_INITIAL_STATE = productsIDs;
const products = [
    {
        id: '123123-234-2341-123123-123123',
        name: 'Greek amphora 1',
        imageUrl: '//c1.staticflickr.com/5/4215/35504896635_ec1a78af43_b.jpg',
        shortDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        longDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut. Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        price: '101K USD'
    },
    {
        id: '123123-12342-456456-123123-123123',
        name: 'Greek amphora 2',
        imageUrl: '//c1.staticflickr.com/5/4215/35504896635_ec1a78af43_b.jpg',
        shortDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        longDesc: 'Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut. Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.Cupidatat irure magna magna in Lorem Lorem est tempor tempor ut.',
        price: '102K USD'
    }];

const mockStore = configureStore()

const initialState = { cart : CART_INITIAL_STATE , allProducts: products, menu : MENU_INITIAL_STATE } ;
    let store, wrapper;  

describe('Cart outer tests', () => {
    

     beforeEach(()=>{
        store = mockStore(initialState);
        wrapper = mount( <Provider store={store}><MemoryRouter><ConnectedCart /></MemoryRouter></Provider> );
    })

    it("has Layout", () => {
        expect(wrapper.find(Layout).length).toBeGreaterThan(0)
    })
    
    it("has HeroArea Component", () => {
        expect(wrapper.find(".App-hero").length).toBeGreaterThan(0)
    })

    it("always renders a section", () => {
        const sections = wrapper.find("section");
        expect(sections.length).toBeGreaterThan(0);
    });
    
    it("contains remove buttons of products", () => {
        const removeButtons = wrapper.find("#remove-button");
        expect(removeButtons.length).toBe(2);
    });



    it(`check DELETE_PRODUCT_FROM_CART`, () => {

        const deleteProductAction = {type: 'DELETE_PRODUCT_FROM_CART', 
                                    productId : '123123-234-2341-123123-123123'};
                                    
        const firstProductDeleteButton = wrapper.find("#remove-button").first();
        firstProductDeleteButton.simulate('click');

        // Test if my store dispatched the expected actions
        const actions = store.getActions().map(action => action.type);
        const expectedPayload = deleteProductAction.type;
        expect(actions).toEqual([expectedPayload]);
    })
})

describe('Cart snapshot tests', () => {

    it('test weird snapshot', () => {

        const store = mockStore(initialState);

        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <ConnectedCart />
                </MemoryRouter>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
 })

