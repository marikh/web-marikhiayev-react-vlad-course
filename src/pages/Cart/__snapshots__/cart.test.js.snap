// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart snapshot tests test weird snapshot 1`] = `
<div
  className="App"
  onClick={[Function]}
>
  <header
    className="App-header"
  >
    <div
      className="App-menubar"
    >
      <label
        className="hamburger-icon fa fa-bars"
        onClick={[Function]}
      />
      <div
        className="menu-content"
      >
        <div
          className="links-container"
        >
          <a
            className="page-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Home
            </span>
          </a>
          <a
            className="page-link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
          <a
            className="page-link"
            href="/products"
            onClick={[Function]}
          >
            <span>
              Products
            </span>
          </a>
          <a
            className="page-link"
            href="/contact"
            onClick={[Function]}
          >
            <span>
              Contact
            </span>
          </a>
          <a
            className="page-link"
            href="/login"
            onClick={[Function]}
          >
            <span>
              SignIn
            </span>
          </a>
        </div>
        <div
          className="socialLinks"
        >
          <div
            className="social-links"
          >
            <a
              className="fa fa-facebook"
              href="facebook.com"
              target="_blank"
            />
            <a
              className="fa fa-github"
              href="github.com"
              target="_blank"
            />
            <a
              className="fa fa-envelope"
              href="vnovick@gmail.comm"
              target="_blank"
            />
          </div>
        </div>
      </div>
      <div
        className="overlay"
      />
    </div>
    <section
      className="App-subheader"
    >
      <div
        className="App-hero"
      >
        <h3
          className="heading heading-3"
        >
          Cart
        </h3>
      </div>
    </section>
  </header>
  <section
    className="App-body"
  >
    <div
      className="raised-2 section"
    >
      <div
        className="products-horizontal-gallery"
      >
        <div
          className="raised-2 card product-card"
          style={
            Object {
              "backgroundImage": "url(undefined)",
            }
          }
        >
          <a
            className="product-wrapping-link"
            href="/products/undefined"
            onClick={[Function]}
          >
            <div
              className="product-top-title"
            >
              <p
                className="product-title"
              />
              <p
                className="product-price"
              />
            </div>
            <p
              className="product-bottom-desc"
            />
          </a>
          <button
            className="cart-product-delete-button"
            id="remove-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
        <div
          className="raised-2 card product-card"
          style={
            Object {
              "backgroundImage": "url(undefined)",
            }
          }
        >
          <a
            className="product-wrapping-link"
            href="/products/undefined"
            onClick={[Function]}
          >
            <div
              className="product-top-title"
            >
              <p
                className="product-title"
              />
              <p
                className="product-price"
              />
            </div>
            <p
              className="product-bottom-desc"
            />
          </a>
          <button
            className="cart-product-delete-button"
            id="remove-button"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </section>
  <footer
    className="App-footer"
  >
    <div
      className="social-links"
    >
      <a
        className="text-link"
        href="facebook.com"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="text-link"
        href="github.com"
        target="_blank"
      >
        Github
      </a>
      <a
        className="text-link"
        href="vnovick@gmail.comm"
        target="_blank"
      >
        Email
      </a>
    </div>
  </footer>
</div>
`;
